data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/http/OAuthAccessTokenManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.http;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.opensearch.dataprepper.plugins.sink.http.configuration.BearerTokenOptions;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Base64;
+import java.util.Map;
+
+public class OAuthAccessTokenManager {
+
+    public static final String BASIC = "Basic ";
+
+    public static final String BEARER = "Bearer ";
+
+    public static final String APPLICATION_X_WWW_FORM_URLENCODED = "application/x-www-form-urlencoded";
+    public static final String EXP = "exp";
+    public static final String ACCESS_TOKEN = "access_token";
+
+    public static final String REFRESH_TOKEN = "refresh_token";
+
+    private final ObjectMapper objectMapper;
+
+    private HttpClientBuilder httpClientBuilder;
+
+
+    public OAuthAccessTokenManager(final HttpClientBuilder httpClientBuilder){
+        this.httpClientBuilder = httpClientBuilder;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public String getAccessToken(final BearerTokenOptions bearerTokenOptions) {
```

Comment:
There are open-source libraries which can handle OAuth2 workflows. Trying to do it ourselves may result in it working only for certain scenarios.

Reply:
Thanks for review @dlvenable , sure I shall use library to handle OAuth2